tracer/src/Datadog.Trace/Util/UriHelpers.cs

```diff
@@ -145,5 +145,20 @@ public static bool IsIdentifierSegment(string absolutePath, int startIndex, int
 
             return containsNumber;
         }
+
+        public static Uri Combine(Uri baseUri, string relativePath)
+        {
+            var builder = new UriBuilder(baseUri);
+            builder.Path =
+                builder.Path.EndsWith("/")
+                    ? (relativePath.StartsWith("/")
+                           ? $"{builder.Path.Substring(0, builder.Path.Length - 1)}{relativePath}"
+                           : $"{builder.Path}{relativePath}")
+                    : (relativePath.StartsWith("/")
```

Comment:
I don't feel very strongly about this, but I would expect a path that starts with `/` to be absolute and replace any existing path in the base uri.

Compare to:
```
System.IO.Path.Combine("a\\b", "c")
    => "a\b\c" ("c" is relative and is appended)
System.IO.Path.Combine("a\\b", "\\c")
    => "\c"  ("\c" is absolute and replaces "a\b")

-----

var baseUri = new Uri("http://host/a/b/");

new Uri(baseUri, "c")
    => "http://host/a/b/c" ("c" is relative and is appended)
new Uri(baseUri, "/c")
    => "http://host/c"  ("/c" is absolute and replaces "a/b")
```

If you want to keep this unconventional behavior, do you mind calling it out in an xml doc comment?

Reply:
I assume this was intentional since you didn't just use
```csharp
Combine(Uri baseUri, string relativePath) => new Uri(baseUri, relativePath)
```

Reply:
Maybe we should use _real_ relative paths by dropping the leading separators in our constants (replace `"/v0.4/traces"` with `"v0.4/traces"`, etc).

Reply:
I have to disagree in this case. In your System.IO.Path example, both "parts" can be either relative or absolute. In that case, appending an absolute path to a relative path makes no sense, so the default behaviour is a best guess.

For this example, the second argument _can't_ be an absolute path, it's a relative path. IMO this is a different situation. I would expect the relative part to be "tacked on the end". This API is trying to solve the fact that

- `http://localhost/a` + `b/c` 
- `http://localhost/a/` + `b/c` 
- `http://localhost/a` + `/b/c` 
- `http://localhost/a/` + `/b/c` 

are all conceptually trying to do the same thing, create the url `http://localhost/a/b/c`. 

Also, the base URI _could_ be customer-provided, so may or may not have a trailing `/`. Having an API like this seems to be the easiest way to avoid strange errors. e.g.

```csharp
// trailing slash means c is appended
new Uri(new Uri("http://host/a/b/"), "c") => "http://host/a/b/c" 

// no trailing slash means last path segment is replaces
new Uri(new Uri("http://host/a/b"), "c") => "http://host/a/c" 
```

This 👆 has bitten me _many_ times with `HttpClient`.

> // "/c" is absolute and replaces "a/b"
> new Uri(baseUri, "/c") => "http://host/c"

IMO, I absolutely do _not_ expect this behaviour either 😉 

> I don't feel very strongly about this...
> If you want to keep this unconventional behavior, do you mind calling it out in an xml doc comment

Given you're not passionate about it, I'll go with this option, and document it extensively 🙂 I'll flag that it's an unconventionally sensible API 😉 😛

Reply:
Just saw your reply now, a week later. Thanks for taking the time to explain in such detail and for adding the code comments!